<launch>
  <machine name="local" address="localhost" default="true" />

  <param name="use_sim_time" value="true"/>
  <arg name="dataset" default="" />
  <arg name="rate" default="1.0" />

  <node name="gMappingSlam" pkg="slam_fmwk" type="gmapping"
        cwd="node" required="true" output="screen">
    <remap from="laser_scan" to="/base_scan" />
    <param name="rbpf_particles_nm" value="30" />
    <param name="async_correction" value="false" />

    <!-- Internal constants for ros-->
    <param name="ros_rviz_map_publishing_rate" value="5.0" />
    <param name="ros_tf_buffer_duration" value="5.0" /> <!-- in seconds-->
    <param name="ros_filter_queue_size" value="1000" />
    <param name="ros_subscribers_queue_size" value="1000" />
  </node>

  <node name="player" pkg="rosbag" type="play"
        args="--delay=3 --clock --rate $(arg rate) $(arg path)" />

  <node name="rviz" pkg="rviz" type="rviz" respawn="true"
        args="-d $(find slam_fmwk)/rviz/debug.rviz" />

</launch>
